pkg/admission/prep_downscale.go

```diff
@@ -468,44 +513,42 @@ func sendPrepareShutdownRequests(ctx context.Context, logger log.Logger, client
 	span, ctx := opentracing.StartSpanFromContext(ctx, "admission.sendPrepareShutdownRequests()")
 	defer span.Finish()
 
-	g, _ := errgroup.WithContext(ctx)
-	for _, ep := range eps {
-		ep := ep // https://golang.org/doc/faq#closures_and_goroutines
-		g.Go(func() error {
-			logger, ctx := spanlogger.New(ctx, logger, "admission.PreparePodForShutdown", tenantResolver)
-			defer logger.Span.Finish()
+	// Attemt to POST to every prepare-shutdown endpoint. If any fail, we'll
+	// need to undo them with a DELETE.
 
-			logger.SetSpanAndLogTag("url", ep.url)
-			logger.SetSpanAndLogTag("index", ep.index)
+	const maxGoroutines = 32
 
-			req, err := http.NewRequestWithContext(ctx, http.MethodPost, "http://"+ep.url, nil)
-			if err != nil {
-				level.Error(logger).Log("msg", "error creating HTTP POST request", "err", err)
-			}
-
-			req.Header.Set("Content-Type", "application/json")
-			req, ht := nethttp.TraceRequest(opentracing.GlobalTracer(), req)
-			defer ht.Finish()
-
-			resp, err := client.Do(req)
-			if err != nil {
-				level.Error(logger).Log("msg", "error sending HTTP POST request", "err", err)
+	g, ectx := errgroup.WithContext(ctx)
+	g.SetLimit(maxGoroutines)
+	for _, ep := range eps {
+		ep := ep
```

Comment:
nit: we don't need this in go 1.22 anymore (we use go 1.22 in rollout-operator). same comment in next for-loop.

Reply:
This makes me a little paranoid as I know others are using this tool. We specify 1.22 in our go.mod but IIUC you can still build and run with an older compiler. And if you don't run the tests it'll probably be race central. :) 
Unless you know something that can lessen my paranoia, I think I'll leave it.

Reply:
> We specify 1.22 in our go.mod but IIUC you can still build and run with an older compiler.

It used to be like that, but now it's mandatory. From https://go.dev/doc/modules/gomod-ref#go
> The go directive sets the minimum version of Go required to use this module. Before Go 1.21, the directive was advisory only; now it is a mandatory requirement: Go toolchains refuse to use modules declaring newer Go versions.